src/facade/cmd_arg_parser.h

```diff
@@ -15,6 +15,22 @@
 
 namespace facade {
 
+// Helper class for numerical range restriction during parsing
+template <auto min, auto max> struct FInt {
+  decltype(min) value = {};
+  operator decltype(min)() {
+    return value;
+  }
+
+  static_assert(std::is_same_v<decltype(min), decltype(max)>, "inconsistent types");
+  static constexpr auto kMin = min;
+  static constexpr auto kMax = max;
+};
```

Comment:
Some nits, 

1. You can use `using Min=decltype(min);` or even better just avoid the `non type parameters` altogether with `typename Min, typename Max`. Generally speaking, non type parameters are concrete instantiations, for f<1, 2> is different than f<2, 2> and their instantiations can easily get out of hand in generic code (although this is not a concern here bit I leave it here for informational purposes).

 2. You don't really need the `implicit conversion operator` below. `ConvertFInt` can just return the type of `min` directly. At the end of the day you just care for the `.value` if the parser has no errors, so returning the underline type/value is more than enough

Reply:
unfortunately ConvertFInt can't return type directly, because I construct tuple and to return direct type I need to do some type deduction that is a little more complex task, but if you want I can implement it